spec/build-rake-spec.coffee

```diff
@@ -31,24 +31,39 @@ describe 'Rakefile provider', ->
       expect(fs.existsSync.calls[0].args[0])
         .toEqual(path.join(projectDir, 'Rakefile'))
 
-    it 'runs rake to get the list of tasks', ->
-      spyOn(child_process, 'exec').andCallFake ->
-        child_process.exec.mostRecentCall.args[2](null, '', '')
-      
-      waitsForPromise -> builder.settings()
-        
-      runs ->
-        expect(child_process.exec.mostRecentCall.args[0]).toEqual('rake -T')
-
     it 'returns the found rake tasks', ->
       # mocked "rake -T" output
       rakeOutput = "rake test   # run all tests\n"
-
       spyOn(child_process, 'exec').andCallFake ->
         child_process.exec.mostRecentCall.args[2](null, rakeOutput, '')
-      
+
       waitsForPromise -> builder.settings().then (settings) ->
         expect(settings.length).toBe(1)
         expect(settings[0].exec).toEqual('rake')
         expect(settings[0].args).toEqual(['test'])
-        
+
+    describe 'on non-Windows', ->
+
+      it 'runs rake to get the list of tasks', ->
+        Object.defineProperty process, 'platform', value: 'MockOS'
+        spyOn(child_process, 'exec').andCallFake ->
+          child_process.exec.mostRecentCall.args[2](null, '', '')
+
+        waitsForPromise -> builder.settings()
+
+        runs ->
+          expect(child_process.exec.mostRecentCall.args[0]).toEqual('rake -T')
+
+    describe 'on Windows', ->
+
+      it 'runs rake to get the list of tasks', ->
```

Comment:
I'd describe the test as `runs rake.bat to get the list of tasks` so the difference from the previous test is more obvious.